src/api/resources/profile.py

```diff
@@ -1,143 +1,47 @@
-
-    def update(self, instance, validated_data):
-        raise exceptions.MethodNotAllowed('UPDATE')
-        # user_data = validated_data.pop('user', None)
-        # user = get_user_model().objects.get(profile=instance)
-        # # user.username = user_data.get('username', instance.username)
-        # # user.email = user_data.get('email', instance.email)
-        # if 'password' in user_data:
-        #     user.set_password(user_data['password'])
-        # user.save()
-        # super(FullProfileSerializer, self).update(instance, validated_data)
-        # return self.Meta.model.objects.get(id=instance.id)  # ensure PATCH returns refreshed data
-
-    class Meta:
-        model = Profile
-        fields = [
-            'id',
-            'password',
-            'email',
-            'first_name',
-            'last_name',
-            'created_on',
-            'updated_on',
-            'updated_by',
-        ]
+        fields = ["id", "created_on", "updated_on", "updated_by"]
 
 
 class ProfileFilterSet(BaseAPIFilterSet):
-    organization = django_filters.UUIDFilter(field_name='projects__project__tagged_items__tag_id', distinct=True,
-                                             label='Associated with organization associated with at least one project '
-                                                   'associated with profile')
-    project = django_filters.UUIDFilter(field_name='projects__project', distinct=True,
-                                        label='Associated with project')
+    organization = django_filters.UUIDFilter(
+        field_name="projects__project__tagged_items__tag_id",
+        distinct=True,
+        label="Associated with organization associated with at least one project "
+        "associated with profile",
+    )
+    project = django_filters.UUIDFilter(
+        field_name="projects__project", distinct=True, label="Associated with project"
+    )
 
     class Meta:
         model = Profile
-        fields = ['organization', 'project', 'email']
-
-
-class UnauthenticatedProfileCreatePermission(permissions.BasePermission):
-    def has_permission(self, request, view):
-        if request.user.is_authenticated:
-            return False
-        else:
-            return request.method == 'POST'
-
-
-class AuthenticatedProfilePermission(permissions.BasePermission):
-    def has_permission(self, request, view):
-        if request.user.is_authenticated:
-            return request.method != 'POST'
-        return False
+        fields = ["organization", "project", "email"]
 
 
 class ProfileViewSet(BaseApiViewSet):
     serializer_class = ProfileSerializer
     queryset = Profile.objects.all()
-    permission_classes = [
-        Or(UnauthenticatedReadOnlyPermission,
-           UnauthenticatedProfileCreatePermission,
-           AuthenticatedProfilePermission
-           )
-    ]
+    permission_classes = [UnauthenticatedReadOnlyPermission]
```

Comment:
What's our intention of setting the permission class to be `UnauthenticatedReadOnlyPermission`?